### ISSUE TITLE
Accept hectad quadrants as spatial references

### ISSUE THREAD

**Issue by sacrevert** [BiologicalRecordsCentre/iRecord#1737]
Whilst the Bryophytes (BBS) iRecord data survey has the text field "Quadrant", actual quadrant grid references are not permitted as spatial entities.

If we are to import historic BBS data to iRecord then we need to amend this so that they are recognised. I don't know how much work this is likely to be, but if it is significant and there is limited time under existing BRC contracts, then it is potentially something that the BBS would consider funding.

**Comment by DavidHepper** [comment]
Do you mean Hectad Quadrants of the form SK54NE 
        or Monad Quadrants of the form SK5143NE 
?

**Comment by sacrevert** [comment]
Hectad quadrants (I've never encountered a monad quadrant in botanical recording tbh)

**Comment by johnvanbreda** [comment]
I think this is less than 2 days work, but I'll need to scan the code for assumptions about the grid reference format based on the length of the string so there's a bit of uncertainty in this estimate.

**Comment by sacrevert** [comment]
@johnvanbreda is there any chance of prioritising this for this summer? We are currently talking to the BLS about the import of historic lichen data, and this issue affects them too.

**Comment by DavidHepper** [comment]
Will entry of hectad quadrants only be enabled per taxon group? We don't want Odonata records arriving with this geograph, please.

**Comment by sacrevert** [comment]
This is for the import of historic data and elsewhere will presumably be configurable as per any other form

**Comment by johnvanbreda** [comment]
@sacrevert @DavidHepper I had assumed that this would be coded by enhancing the current support for OSGB grid references to detect grid references of the format [A-Z][A-Z][0-9][0-9][A-Z][A-Z] and treat them as hectads. This would mean that anywhere that OSGB grids are enabled, the hectad format would be recognised, so there would be nothing to stop Odonata records using this format when imported. That is how the current DINTY support is handled. Given David's concern about not using this format inappropriately, would it be better if this format was completely separate so it would only be available on data entry forms when "OSGB Hectads" was selected as an available input format, plus imports would only use this format if explicitly selected. This does mean that an import which mixes standard OSGB and hectads would need a column defining which row is which format (similar to an import which has mixed map reference types in the current importer). Let me know your thoughts.

**Comment by sacrevert** [comment]
I don't really see how folk would accidentally enter this format; it could be further constrained to ensure that suffix letters are actual quadrants rather than any letter combinations. Given that it is a standard historic format (at least for botanical and fungal data) I would prefer it to be implemented in the same way as tetrad DINTY, i.e. with minimal extra complications.

Sent from Outlook for Android<https://aka.ms/AAb9ysg>
________________________________

**Comment by DavidRoy** [comment]
I agree with Oli. This is a well-established format in the biological recording community.  I would favour making this a generic improvement across all current support for OSGB (and OSI) grid reference entry. If, in the unlikely event, that we get a lot of unintended uses then we can revisit for the extra complexity of making this configurable

**Comment by sacrevert** [comment]
[heart]         Oliver Pescott reacted to your message:
________________________________

**Comment by johnvanbreda** [comment]
A quick question - if a record is recorded at a point, with a coordinate uncertainty of between 1000 and 2000m, then the output grid reference calculated for reporting outputs uses the DINTY format for OSGB grids. If a point has a coordinate certainty of > 2000 and <= 5000m, should we use the quadrant? Or use just the 10km as now for output grid references so that quadrants are just for input and imports?

**Comment by kitenetter** [comment]
From an entomological point of view I'd prefer not to use the quadrants, as I'm not aware that they are in use for most ento recording schemes, and are likely to add confusion/complexity for verifiers. But Oli may have a different view from the botanical perspective.

**Comment by sacrevert** [comment]
No, I don't think so. To be clear, hectad quadrants are not in active _current_ use for botanical recording either; the main point here is to allow for the upload or other entry of historic data resolved to this scale.

**Comment by johnvanbreda** [comment]
Thanks both. That's as I'd expected. The coding is done ready for the next warehouse release.
